File: mobile/src/types/styles.ts
/**
 * Type definitions for artistic style transfer
 */

export type StyleTier = 'free' | 'premium';

export type StyleCategory =
  | 'abstract'
  | 'watercolor'
  | 'oil_painting'
  | 'geometric'
  | 'cosmic'
  | 'nature'
  | 'pop_art'
  | 'minimalist';

export interface StylePreset {
  id: string;
  name: string;
  displayName: string;
  description: string;
  category: StyleCategory;
  tier: StyleTier;
  thumbnailUrl: string | null;
  sortOrder: number;
}

export interface StyleListResponse {
  free: StylePreset[];
  premium: StylePreset[];
}

export type StyleJobStatus = 'pending' | 'processing' | 'completed' | 'failed';

export interface StyleJob {
  id: string;
  status: StyleJobStatus;
  progress: number;
  currentStep: string | null;
  previewUrl: string | null;
  resultUrl: string | null;
  stylePreset: StylePreset;
  processingTimeMs: number | null;
  errorType: string | null;
  errorMessage: string | null;
  createdAt: string;
  websocketUrl: string | null;
}

export interface StyleJobSubmitRequest {
  photoId: string;
  stylePresetId: string;
  processingJobId?: string;
}

export interface StyleJobListResponse {
  items: StyleJob[];
  total: number;
}

File: backend/app/schemas/styles.py
"""Pydantic schemas for style transfer API."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class StylePresetResponse(BaseModel):
    """Response schema for a single style preset."""

    id: UUID
    name: str
    display_name: str
    description: str
    category: str
    tier: str
    thumbnail_url: str | None = None
    sort_order: int

    model_config = {"from_attributes": True}


class StyleListResponse(BaseModel):
    """Response schema for grouped style presets."""

    free: list[StylePresetResponse]
    premium: list[StylePresetResponse]


class StyleJobSubmitRequest(BaseModel):
    """Request schema for submitting a style transfer job."""

    photo_id: UUID = Field(..., description="Photo ID to apply style to")
    style_preset_id: UUID = Field(..., description="Style preset to apply")
    processing_job_id: UUID | None = Field(
        None, description="Optional processing job ID to use processed iris result"
    )


class StyleJobResponse(BaseModel):
    """Response schema for style job status."""

    id: UUID
    status: str
    progress: int
    current_step: str | None = None
    preview_url: str | None = None
    result_url: str | None = None
    style_preset: StylePresetResponse
    processing_time_ms: int | None = None
    error_type: str | None = None
    error_message: str | None = None
    created_at: datetime
    websocket_url: str | None = None

    model_config = {"from_attributes": True}


class StyleJobListResponse(BaseModel):
    """Response schema for paginated list of style jobs."""

    items: list[StyleJobResponse]
    total: int
